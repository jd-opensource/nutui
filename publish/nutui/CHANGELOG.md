## v4.0.1

`2023-02-10`

NutUI 非常感谢您对开源事业做出的贡献！🌷🌷🌷
本次社区贡献者 @dongj0316 @mikasayw @rocwong-cn @thx125 @gyt95

* :zap: feat(checkbox): add checkbox new props of shape (#1978) @mikasayw
* :zap: feat(tabs): title bar supports automatic scrolling (#2035) @gyt95
* :zap: feat: calendar 添加自定义方法 (#2084) @lkjh3214
* :zap: feat: 增加 unplugin 插件配置导出 (#2097)  @eiinu
* :zap: feat: 日历组件新增 周选择功能 (#2102)   @lkjh3214
* :zap: feat(input): 新增slots (#2098)  @lkjh3214
* :zap: feat(badge): 新增bubble样式 (#2091) @dongj0316
* :bug: fix(list): list 虚拟列表未设置height，滚动加载有问题 #2088 @szg2008
* :bug: fix(list): 虚拟列表在ios手机上，滚动条滚动到顶部会出现白屏。 #2031 @szg2008
* :bug: fix: barrage，sinature demo修改 @Ymm0008
* :bug: fix(address): 地址问题修改 (#2054)  @yangxiaolu1993
* :bug: fix: drag，configprovide，badge 修改demo @Drjingfubo
* :bug: fix: taro h5 demo修改 @szg2008
* :bug: fix: jdt-style error @Drjingfubo
* :bug: fix: sku组件修改 (#2056)  @yangxiaolu1993
* :bug: fix: collapse,textarea 优化  @Ymm0008
* :bug: fix(tour): demo style edit @Drjingfubo
* :bug: fix(menu): change style @Drjingfubo
* :bug: fix(calendar): demo样式修改，文档修改 (#2052) @lkjh3214
* :bug: fix(config-provider): 覆盖nut-icon属性 (#2022) @thx125
* :bug: fix(table): demo optimization @Drjingfubo
* :bug: fix: imagepreview问题修改 (#2068) @yangxiaolu1993
* :bug: fix: 修复uploader组件的is-deletable在列表模式下不生效 (#2073) @rocwong-cn
* :bug: fix(input): 事件触发修改 (#2081) @lkjh3214
* :bug: fix(notify): resolve teleport warn @Drjingfubo
* :bug: fix(uploader): 修复图片拉伸样式问题 #2026 @richard1015
* :bug: fix: change sideEffects & default export (#2094) @eiinu
* :bug: fix: 公告栏 noticebar 插槽使用时不更新问题修改 (#2104) @yangxiaolu1993
* :bug: fix: 公告栏demo修改 (#2105) @yangxiaolu1993
* :bug: fix: timer ts error resolve @szg2008
* :bug: fix: 签名组件交互优化 @Ymm0008
* 📖 docs(addresslist): taro env use edit  @richard1015
* 📖 docs: popover修改 (#2053) @yangxiaolu1993
* 📖 docs: price 文档问题 (#2064)  @ailululu
* 📖 docs: card edit @Drjingfubo
* 📖 docs(tabbar): taro add punctuation @Drjingfubo
* 📖 docs: 新增4.0介绍 (#2086)  @yangxiaolu1993

## v4.0.0 🚀

`2023-01-12`

### 新特性

- :zap: 全新的升级，基于 vite4 构建
- :sparkles: 轻量化 安装体积由 14MB 下降至 8MB，减少了 39%
- :art: 支持动态定制主题
- :sparkles: 专属图标库`nutui-icons`
- :sparkles: 支持`unplugin-vue-components`插件
- :sparkles: 新增组件`Tour``PullRefresh`

> 完整内容请移步：「[4.0更新介绍](https://nutui.jd.com/next)」

小程序版本依赖 Taro ^3.5.10

NutUI 非常感谢您对开源事业做出的贡献！🌷🌷🌷

本次社区贡献者 @kasonyang @gyt95 @thx125 @Coffee-C @sjx12 @mikasayw @jxt776

### ⚠️ 升级必读

- [从 v3 升级到 v4](https://nutui.jd.com/4x/index.html#/zh-CN/guide/migrate-from-v3)
