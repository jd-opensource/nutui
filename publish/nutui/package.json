{
  "name": "@nutui/nutui",
  "version": "4.0.0-beta.7",
  "description": "京东风格的轻量级移动端 Vue2、Vue3 组件库（支持小程序开发）",
  "main": "dist/nutui.umd.cjs",
  "module": "dist/nutui.es.js",
  "style": "dist/style.css",
  "type": "module",
  "typings": "dist/types/index.d.ts",
  "sideEffects": [
    "dist/packages/_es/*",
    "dist/packages/**/style*"
  ],
  "vetur": {
    "tags": "dist/smartips/tags.json",
    "attributes": "dist/smartips/attributes.json"
  },
  "web-types": "dist/smartips/web-types.json",
  "keywords": [
    "nutui",
    "nutui2",
    "nutui3",
    "nutui4",
    "vue",
    "vue3",
    "webpack",
    "vue component",
    "vue3 component",
    "vite",
    "jdc",
    "jdcfe"
  ],
  "author": "jdcfe",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org/"
  },
  "scripts": {
    "publish:beta": "npm publish --tag=beta"
  },
  "standard-version": {
    "scripts": {
      "postchangelog": "node ./jd/updateChangelog.js"
    }
  },
  "dependencies": {
    "@nutui/icons-vue": "0.0.19",
    "sass": "^1.50.0",
    "vscode": "^1.1.37",
    "vue-router": "^4.0.12"
  },
  "devDependencies": {
    "@commitlint/cli": "^10.0.0",
    "@commitlint/config-conventional": "^10.0.0",
    "@tarojs/cli": "3.5.6",
    "@tarojs/mini-runner": "3.5.6",
    "@tarojs/plugin-framework-vue3": "3.5.6",
    "@tarojs/plugin-html": "3.5.6",
    "@tarojs/plugin-platform-weapp": "3.5.6",
    "@tarojs/runtime": "3.5.6",
    "@tarojs/taro": "3.5.6",
    "@tarojs/webpack5-runner": "3.5.6",
    "@types/jest": "^26.0.22",
    "@types/node": "^17.0.16",
    "@typescript-eslint/eslint-plugin": "^4.20.0",
    "@typescript-eslint/parser": "^4.20.0",
    "@vitejs/plugin-vue": "^4.0.0",
    "@vue/compiler-dom": "^3.2.45",
    "@vue/compiler-sfc": "^3.2.45",
    "@vue/eslint-config-prettier": "^6.0.0",
    "@vue/eslint-config-typescript": "^7.0.0",
    "@vue/test-utils": "^2.0.0-rc.18",
    "autoprefixer": "^10.3.4",
    "axios": "^0.21.0",
    "babel-preset-taro": "3.5.6",
    "canvas": "^2.9.0",
    "codesandbox": "^2.2.3",
    "eslint": "^7.23.2",
    "eslint-config-taro": "3.5.6",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-vue": "^7.8.0",
    "eslint-visitor-keys": "2",
    "fs-extra": "^9.1.0",
    "highlight.js": "^10.3.1",
    "husky": "^6.0.0",
    "inquirer": "^8.2.0",
    "jest": "^26.6.3",
    "lint-staged": "^10.5.0",
    "lzutf8": "0.6.0",
    "markdown-it": "^13.0.1",
    "markdown-it-container": "^3.0.0",
    "prettier": "^2.0.0",
    "remark-codesandbox": "^0.10.1",
    "rollup-plugin-visualizer": "^5.8.3",
    "standard-version": "^9.3.0",
    "swiper": "6.5.1",
    "transliteration": "^2.2.0",
    "ts-jest": "^26.5.5",
    "typescript": "^4.9.3",
    "vite": "^4.0.0",
    "vite-plugin-md": "^0.21.5",
    "vue": "^3.2.24",
    "vue-jest": "^5.0.0-alpha.7",
    "vue-tsc": "^1.0.9",
    "webpack": "^5.69.0"
  },
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true
    },
    "extends": [
      "plugin:vue/vue3-essential",
      "eslint:recommended",
      "@vue/typescript/recommended",
      "@vue/prettier",
      "@vue/prettier/@typescript-eslint"
    ],
    "parserOptions": {
      "ecmaVersion": 2020
    },
    "rules": {
      "vue/no-unused-components": "off",
      "no-debugger": "off"
    }
  },
  "engines": {
    "node": ">= 16.0.0",
    "pnpm": ">= 7.0.0"
  },
  "lint-staged": {
    "*.md": "prettier --write",
    "*.{ts,tsx,js,vue,scss}": "prettier --write"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/jdf2e/nutui.git"
  }
}