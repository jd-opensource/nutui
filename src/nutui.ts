export default {
  async install(app: any): Promise<void> {
    /** webpack */
    // const files = require.context('@/packages', true, /index\.vue$/);
    // files.keys().forEach(component => {
    //   const componentEntity = files(component).default;
    //   app.component(componentEntity.name, componentEntity);
    // });

    /** vite */
    const modules = import.meta.globEager('/src/packages/**/index.vue');
    for (const key in modules) {
      const componentEntity = modules[key]?.default;
      app.component(componentEntity.name, componentEntity);
    }
  }
};
