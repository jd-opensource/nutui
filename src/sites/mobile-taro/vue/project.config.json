{
  "miniprogramRoot": "./dist",
  "projectname": "@nutui/nutui-taro-mobile",
  "description": "nutui-taro-vue",
  "appid": "touristappid",
  "setting": {
    "urlCheck": true,
    "es6": false,
    "postcss": false,
    "minified": false
  },
  "compileType": "miniprogram"
}
