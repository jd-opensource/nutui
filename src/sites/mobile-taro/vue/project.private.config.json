{
  "setting": {},
  "condition": {
    "plugin": {
      "list": []
    },
    "game": {
      "list": []
    },
    "gamePlugin": {
      "list": []
    },
    "miniprogram": {
      "list": [
        {
          "name": "business/pages/address/index",
          "pathName": "business/pages/address/index",
          "query": "",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/noticebar/index",
          "pathName": "feedback/pages/noticebar/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/progress/index",
          "pathName": "feedback/pages/progress/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/swiper/index",
          "pathName": "feedback/pages/swiper/index",
          "query": "",
          "scene": null
        }
      ]
    }
  }
}