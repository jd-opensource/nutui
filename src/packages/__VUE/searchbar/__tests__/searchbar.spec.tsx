import { mount } from '@vue/test-utils'
import { Searchbar } from '@nutui/nutui'
import { ref } from 'vue'

test('Searchbar: basic usage', () => {
  const wrapper = mount(() => <Searchbar modelValue="123" />)
  expect(wrapper.find('.nut-searchbar').html()).toMatchSnapshot()
})

test('Searchbar: should limit maxlength of input value when using maxlength prop', async () => {
  const val = ref('9999')
  const wrapper = mount(() => <Searchbar maxLength={3} v-model={val.value} />)

  const input = wrapper.find('input')
  input.trigger('input')
  expect(val.value).toBe('999')
})

test('Searchbar: should format input value when type is number', () => {
  const val = ref('')
  const wrapper = mount(() => <Searchbar inputType="number" v-model={val.value} />)
  const input = wrapper.find('input')
  input.element.value = '1'
  input.trigger('input')
  expect(val.value).toBe('1')

  input.element.value = '12'
  input.trigger('input')
  expect(val.value).toBe('12')
})

test('Searchbar: backgrounds prop test', () => {
  const wrapper = mount(() => {
    return <Searchbar modelValue="" background="red" inputBackground="green" />
  })
  const outer = wrapper.find('.nut-searchbar')
  const inner = wrapper.find('.nut-searchbar__search-input')
  expect(outer.attributes('style')).includes('background: red')
  expect(inner.attributes('style')).includes('background: green')
})

test('Searchbar: change event test', async () => {
  const change = vi.fn()
  const wrapper = mount(() => <Searchbar modelValue="" onChange={change} />)

  const input = wrapper.find('input')
  await input.trigger('input')
  expect(change).toBeCalled()
})

test('Searchbar: focus event focus', async () => {
  const focus = vi.fn()
  const wrapper = mount(() => {
    return <Searchbar modelValue="" onFocus={focus} />
  })

  const input = wrapper.find('input')
  await input.trigger('focus')
  expect(focus).toBeCalled()
})

test('Searchbar: blur event test', async () => {
  const blur = vi.fn()
  const wrapper = mount(() => {
    return <Searchbar modelValue="" onBlur={blur} />
  })

  const input = wrapper.find('input')
  await input.trigger('blur')
  expect(blur).toBeCalled()
})

test('Searchbar: clear event test', async () => {
  const val = ref('3')
  const wrapper = mount(() => <Searchbar v-model={val.value} />)
  const input = wrapper.find('input')
  const clear = wrapper.find('.nut-searchbar__input-clear')
  wrapper.find('input').trigger('input')
  expect(input.element.value).toBe('3')
  await clear.trigger('click')
  expect(val.value).toBe('')
  expect(clear.exists()).toBe(true)
})

test('Searchbar: props shape', async () => {
  const wrapper1 = mount(() => <Searchbar shape="square" />)
  const wrapper2 = mount(() => <Searchbar />)
  const input1 = wrapper1.find('.nut-searchbar__search-input')
  const input2 = wrapper2.find('.nut-searchbar__search-input')
  expect(input1.html()).includes('square')
  expect(input2.html()).includes('round')
})

test('Searchbar: slots', async () => {
  const wrapper = mount(() => (
    <Searchbar>
      {{
        leftout: 'slot leftout',
        leftin: 'slot leftin',
        rightout: 'slot rightout',
        rightin: 'slot rightin'
      }}
    </Searchbar>
  ))
  const leftout = wrapper.find('.nut-searchbar__left-search-icon')
  expect(leftout.html()).includes('slot leftout')

  const leftin = wrapper.find('.nut-searchbar__iptleft-search-icon')
  expect(leftin.html()).includes('slot leftin')

  const rightout = wrapper.find('.nut-searchbar__right-search-icon')
  expect(rightout.html()).includes('slot rightout')

  const rightin = wrapper.find('.nut-searchbar__iptright-search-icon')
  expect(rightin.html()).includes('slot rightin')
})
