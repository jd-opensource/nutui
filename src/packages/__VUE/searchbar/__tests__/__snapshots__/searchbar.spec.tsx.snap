// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Searchbar: basic usage 1`] = `
"<view class="nut-searchbar">
  <!--v-if-->
  <!--v-if-->
  <view class="nut-searchbar__search-input round">
    <!--v-if-->
    <view class="nut-searchbar__input-inner">
      <form class="nut-searchbar__input-form" action="#"><input class="nut-searchbar__input-bar nut-searchbar__input-bar_clear" type="text" maxlength="9999" placeholder="请输入" style="text-align: left;"></form>
    </view>
    <view class="nut-searchbar__input-inner-icon">
      <view class="nut-searchbar__search-icon nut-searchbar__input-clear"><svg class="nut-icon nut-icon-circle-close" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" role="presentation">
          <path d="M511.492 73.063a438.425 438.425 0 0 1 310.545 748.953 439.16 439.16 0 1 1-621.09-621.073A435.695 435.695 0 0 1 511.492 73.046m0-73.063C267.49.018 57.468 172.408 9.87 411.717-37.713 651.042 90.39 890.658 315.823 984.03c225.45 93.355 485.444 14.472 621.005-188.416 135.543-202.906 108.885-473.293-63.659-645.82A509.645 509.645 0 0 0 511.492.002zm-175.7 635.495 299.673-299.674c12.049-12.049 24.115-12.049 36.164 0l15.496 15.497c12.05 12.066 12.05 24.115 0 36.18L387.47 687.122c-12.066 12.05-24.115 12.05-36.181 0l-15.497-15.496c-12.049-12.066-12.049-24.115 0-36.182zm0-284.177 15.496-15.497c12.066-12.049 24.115-12.049 36.18 0l299.64 299.674c12.05 12.049 12.05 24.098 0 36.164l-15.496 15.496c-12.05 12.05-24.115 12.05-36.182 0L335.808 387.465c-12.049-12.05-12.049-24.098 0-36.164z" fill="currentColor" fill-opacity="0.9"></path>
        </svg></view>
      <!--v-if-->
    </view>
  </view>
  <!--v-if-->
</view>"
`;
